tests/hf/test_tapering.py

```diff
@@ -499,3 +456,89 @@ def test_transform_hamiltonian(symbols, geometry, generator, paulix_ops, paulix_
     for i, term in enumerate(sorted_terms):
         assert np.allclose(term[0], ham_ref.terms[0][i])
         assert term[1].compare(ham_ref.terms[1][i])
+
+
+@pytest.mark.parametrize(
+    ("symbols", "geometry", "result"),
+    [
+        (
+            ["H", "H"],
+            np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.40104295]], requires_grad=False),
+            -1.137270174,
+        ),
+    ],
+)
+def test_sector_energy(symbols, geometry, result):
+    r"""Test that _sector_energy returns the correct result."""
+    mol = qml.hf.Molecule(symbols, geometry)
+    hamiltonian = qml.hf.generate_hamiltonian(mol)()
+    energy = _sector_energy(hamiltonian)
+    assert np.isclose(energy, result)
+
+
+@pytest.mark.parametrize(
+    ("symbols", "geometry", "generators", "paulix_ops", "result"),
+    [
+        (
+            ["H", "H"],
+            np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.40104295]], requires_grad=False),
+            [
+                qml.Hamiltonian([1.0], [qml.PauliZ(0) @ qml.PauliZ(1)]),
+                qml.Hamiltonian([1.0], [qml.PauliZ(0) @ qml.PauliZ(2)]),
+                qml.Hamiltonian([1.0], [qml.PauliZ(0) @ qml.PauliZ(3)]),
+            ],
+            [qml.PauliX(wires=[1]), qml.PauliX(wires=[2]), qml.PauliX(wires=[3])],
+            ([1, -1, -1], -1.137270174),
+        ),
+    ],
+)
+@pytest.mark.parametrize(("brute_force", "num_electrons"), [(True, 0), (False, 2)])
```

Comment:
We probably should get rid of this, especially if we decide not to perform the brute force method.